share/shwap/p2p/bitswap/sample_block.go

```diff
@@ -86,7 +86,7 @@ func (sb *SampleBlock) Marshal() ([]byte, error) {
 }
 
 func (sb *SampleBlock) Populate(ctx context.Context, eds eds.Accessor) error {
-	smpl, err := eds.Sample(ctx, sb.ID.RowIndex, sb.ID.ShareIndex)
+	smpl, err := eds.Sample(ctx, shwap.SampleIndex(sb.ID.ShareIndex))
```

Comment:
ShareIndex is only the column

Reply:
The data structure we work with is a table of Rows and Columns. Cells in the table are called shares. Every share can be addressed by coordinates or index. Coordinate in 2D space and index in 1D vector space.

In this case ShareIndex is ColumnIndex(I know it might be confusing in this context), so 
(RowIndex; ShareIndex) is a coordinate.

SampleIndex is an index in 1D space, so we cant use ShareIndex ad SampleIndex.

P.S. Sample is Share + Proof, so in the context of positiom Share and Sample can be used interchangeably.

Reply:
Updated.